src/main/kotlin/com/ampnet/mailservice/service/impl/LinkResolver.kt

```diff
@@ -0,0 +1,27 @@
+package com.ampnet.mailservice.service.impl
+
+import com.ampnet.mailservice.config.ApplicationProperties
+import com.ampnet.mailservice.enums.WalletType
+import java.net.URL
+
+class LinkResolver(applicationProperties: ApplicationProperties) {
```

Comment:
```suggestion
private val baseUrl = URL(applicationProperties.mail.baseUrl).toString()
    private val confirmationPath = applicationProperties.mail.confirmationPath
    private val resetPasswordPath = applicationProperties.mail.resetPasswordPath
    private val newWalletPath = applicationProperties.mail.newWalletPath
    val organizationInvitesLink = baseUrl + applicationProperties.mail.organizationInvitationsPath
    val manageWithdrawalsLink = baseUrl + applicationProperties.mail.manageWithdrawalsPath

    fun getConfirmationLink(token: String): String = "$baseUrl$confirmationPath?token=$token"
    fun getResetPasswordLink(token: String): String = "$baseUrl$resetPasswordPath?token=$token"

    fun getNewWalletLink(walletType: WalletType): String =
        when (walletType) {
            WalletType.USER -> "$baseUrl$newWalletPath/users"
            WalletType.PROJECT -> "$baseUrl$newWalletPath/projects"
            WalletType.ORGANIZATION -> "$baseUrl$newWalletPath/groups"
        }
```
Remove unnecessary back and forth toString(), the first variable baseUrl is put into URL class only for checking if it is valid URL.